apps/root/src/pages/aggregator/swap-container/components/token-picker/index.tsx

```diff
@@ -28,74 +24,52 @@ const AggregatorTokenPicker = ({ shouldShow, onChange, onClose, modalTitle }: Ag
   const tokenList = useTokenList({ chainId: currentNetwork.chainId, curateList: true });
   const customTokens = useCustomTokens(currentNetwork.chainId);
   const isLoadingLists = useIsLoadingAllTokenLists();
-  const dispatch = useAppDispatch();
   const trackEvent = useTrackEvent();
   const { balances, isLoadingBalances, isLoadingPrices } = useWalletBalances(
     activeWallet?.address as Address,
     currentNetwork.chainId
   );
+  const { handleCustomTokenBalances, isLoadingCustomToken } = useHandleCustomTokenBalances();
 
-  const [customTokenToSearch, setCustomTokenToSearch] = React.useState<string | undefined>(undefined);
-
-  const [customToken, isLoadingCustomToken] = useCustomToken(customTokenToSearch);
-
-  let tokens: TokenWithBalance[] = parseTokensForPicker({ tokenList, balances, customTokens }).map(
-    (tokenWithBalance) => ({
-      ...tokenWithBalance,
-      token: {
-        ...tokenWithBalance.token,
-        icon: <TokenIcon token={tokenWithBalance.token} size={8} />,
-      },
-    })
+  const tokens = React.useMemo<TokenWithBalance[]>(
+    () =>
+      parseTokensForPicker({ tokenList, balances, customTokens }).map((tokenWithBalance) => ({
+        ...tokenWithBalance,
+        token: {
+          ...tokenWithBalance.token,
+          icon: <TokenIcon token={tokenWithBalance.token} size={8} />,
+        },
+      })),
+    [tokenList, balances, customTokens]
   );
 
-  const parsedCustomToken: TokenWithBalance | undefined = customToken && {
-    token: {
-      ...customToken.token,
-      icon: <TokenIcon token={customToken.token} size={8} />,
-    },
-    isCustomToken: true,
-    balance:
-      (customToken.balance && {
-        amount: customToken.balance,
-        amountInUnits: formatCurrencyAmount({ amount: customToken.balance, token: customToken.token }),
-        amountInUSD:
-          (customToken.balanceUsd &&
-            parseFloat(formatUnits(customToken.balanceUsd, customToken.token.decimals + 18)).toFixed(2)) ||
-          undefined,
-      }) ||
-      undefined,
+  // TODO: Track custom tokens addition like it was tracked before
```

Comment:
lets add this :)

Reply:
I have 😄 but forgot to remove this comment (done in line 55 of this file)